File: src/main/java/com/model/annotation/XSecondsPastOrPresent.java
package com.model.annotation;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.ElementType.PARAMETER;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target({FIELD, PARAMETER})
@Retention(RUNTIME)
@Constraint(validatedBy = XSecondsPastOrPresentImpl.class)
public @interface XSecondsPastOrPresent {
    public abstract int seconds() default 60;
    public abstract String message() default "Message";
    public abstract Class<?>[] groups() default {};
    public abstract Class<? extends Payload>[] payload() default {};
}

File: src/main/kotlin/com/model/annotation/XSecondsPastOrPresent.kt
package com.model.annotation

import kotlin.annotation.Retention
import javax.validation.Constraint
import kotlin.reflect.KClass
import javax.validation.Payload

@Target(AnnotationTarget.FIELD, AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [XSecondsPastOrPresentImpl::class])
annotation class XSecondsPastOrPresent(
    val seconds: Int = 60,
    val message: String = "Message",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)